### PR TITLE
Generate Note model, implement migration, and include Ruby Singleton …

### LINKED ISSUE #14: Create `NoteRepository` repository

### PR DESCRIPTION
**Pull request by HeLooksAngry**
…module on Repo

Closes studio-thrash/notary#14
Closes studio-thrash/notary#11

### CODE REVIEW

File: lib/notary/entities/note.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class Note < Hanami::Entity
+end
```
Comment (parkersprouse) [inline lib/notary/entities/note.rb:2]: Do we wanna stick to the convenience method style that I used in the User Entity? https://github.com/studio-thrash/notary/blob/main/lib/notary/entities/user.rb

I think we should either do it for all of them, or none of them

File: db/migrations/20220511011526_create_notes.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    # sequel gem extension for enum type
+    extension :pg_enum
+    create_enum :note_type, %w[basic list]
+
+    # from hanami doc on UUID https://guides.hanamirb.org/v1.3/repositories/postgresql/
+    execute 'CREATE EXTENSION IF NOT EXISTS "uuid-ossp"'
+
+    create_table :notes do
+      # primary key definition 
+      column :id, :uuid, null: false, primary_key: true, default: Hanami::Model::Sql.function(:uuid_generate_v4)
```
Comment (parkersprouse) [inline db/migrations/20220511011526_create_notes.rb:12]: You can do this using the `primary_key` helper if you'd prefer, either way works for me though

https://github.com/studio-thrash/notary/blob/main/db/migrations/20220511050104_create_users.rb#L6

File: db/migrations/20220511011526_create_notes.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      column :body, :text, null: false
+      column :type, :note_type, null: false
+
+      column :created_at, DateTime, null: false
+      column :updated_at, DateTime, null: false
+    end
+  end
+
+  down do
+    extension :pg_enum
```
Comment (parkersprouse) [inline db/migrations/20220511011526_create_notes.rb:23]: I haven't read anything about the `extension` helper, does simply calling it again the same way in the `down` method automatically make it remove the extension?

File: db/migrations/20220511011526_create_notes.rb (hunk 2 above)
Comment (HeLooksAngry) [reply db/migrations/20220511011526_create_notes.rb:12]: I'll change it so the migrations share syntax.

File: lib/notary/entities/note.rb (hunk 1 above)
Comment (HeLooksAngry) [reply lib/notary/entities/note.rb:2]: We'll roll all of them for  right now.

File: db/migrations/20220511011526_create_notes.rb (hunk 3 above)
Comment (HeLooksAngry) [reply db/migrations/20220511011526_create_notes.rb:23]: This extension is just for having access to the sequel enum methods. I don't believe it affects any sql ops by itself.

**Review by parkersprouse — APPROVED** [review]

**Pull request merged**